src/fits.cc

```diff
@@ -66,10 +66,28 @@ std::string makeLimitedFitsHeaderImpl(std::vector<std::string> const &paramNames
         } else if (type == typeid(int)) {
             out += (boost::format("%20d") % metadata.get<int>(name)).str();
         } else if (type == typeid(double)) {
-            // use G because FITS wants uppercase E for exponents
-            out += (boost::format("%#20.17G") % metadata.get<double>(name)).str();
+            double value = metadata.get<double>(name);
+            if (!std::isnan(value)) {
+                // use G because FITS wants uppercase E for exponents
+                out += (boost::format("%#20.17G") % value).str();
+            } else {
+                LOGLS_WARN("afw.fits",
+                           boost::format("In %s, found NaN in metadata item '%s'") %
+                                         BOOST_CURRENT_FUNCTION % name);
+                // Convert it to FITS undefined
+                out += " ";
```

Comment:
Is this FITS undefined, or is it `"BLANK   "`? https://heasarc.gsfc.nasa.gov/docs/fcg/standard_dict.html
It could be that there's another translation layer somewhere else in the bowels of afw, or that I'm misreading something.

Reply:
FITS undefined (there is a keyword and an `=` sign)